CalendarApp/BookingRoom/Controllers/PostEventController.cs

```diff
@@ -1,26 +1,40 @@
-﻿using BookingRoom.Models.GoogleCalendar;
+﻿using System.Collections.Generic;
+using BookingRoom.Models.GoogleCalendar;
 using System.Net;
 using System.Net.Http;
 using System.Web.Http;
 using BookingRoom.Models.GoogleEvent;
 using GoogleCalendarService;
 using BookingRoom.Filters;
+using Google.Apis.Calendar.v3;
+using Google.Apis.Calendar.v3.Data;
+using GoogleCalendarService.Manager;
 
 namespace BookingRoom.Controllers
 {
     [GoogleExceptionFilter]
     public class PostEventController : BaseEventController
     {
-        public PostEventController(BookingService bookingService)
-            : base(bookingService)
+        public PostEventController(BookingService bookingService, EventManager manager, CalendarService service)
+            : base(bookingService, manager,service)
         {
 
         }
-        [HttpPost]
-        public HttpResponseMessage EventPost(CalendarEvent eventPost)
+        public HttpResponseMessage PostEvent(CalendarEvent eventPost)
         {
-            BookingService.PostEvent(ToEventConverter.ToEvent(eventPost), eventPost.CalendarID);
-            return Request.CreateResponse(HttpStatusCode.Created, eventPost);
+            Event eventForSearchConflicts = ToEventConverter.ToEvent(eventPost, TimeZone);
+            List<Event> conflictedEvents = EventManager.CheckEvent(eventPost.CalendarID, eventForSearchConflicts, eventPost.ForcedEvent);
+            if (conflictedEvents.Count != 0)
+            {
+                var failRequest = new {status = "Unsuccess", ConflictedList = conflictedEvents};
+                return Request.CreateResponse(HttpStatusCode.OK, failRequest);
+            }
+            else
+            {
+                BookingService.PostEvent(ToEventConverter.ToEvent(eventPost, TimeZone), eventPost.CalendarID);
+            }
+            var passRequest = new { status = "Success", ConflictedList = conflictedEvents };
```

Comment:
But what did you mean when you said that we need to make a response in a same fashion anywhere??